src/easier68k/core/util/parsing.py

```diff
@@ -31,6 +34,220 @@ def parse_literal(literal: str):
+    """
+    Removes all comments from a line (basically makes this line into the 'compiler' version)
+
+    >>> strip_comments('label TRAP #15 * This does a thing')
+    'label TRAP #15 '
+
+    >>> strip_comments('    ORG start ;label')
+    '    ORG start '
+
+    >>> strip_comments(';    ADD D0, D1 * asdf')
+    ''
+
+    :param line: The line to strip comments from
+    :return: The stripped line
+    """
+    to_return = ''
+    for c in line:
+        if c == ';' or c == '*':
+            break
+
+        to_return += c
+
+    return to_return
+
+
+def has_label(line: str) -> bool:
+    """
+    Returns whether or not this line has a label in it (basically if it starts with a space or not)
+    :param line: The line to test
+    :return: Whether the test line has a label in it
+
+    >>> has_label('data    DC.B $A3')
+    True
+
+    >>> has_label('    BEQ test')
+    False
+
+    >>> has_label('  TRAP #15')
+    False
+
+    >>> has_label(';start EQU $400')
+    False
+
+    """
+    stripped = strip_comments(line)
+    if not stripped.strip():  # The line is literally empty after removing comments
+        return False
+    return not strip_comments(line).startswith(' ')
+
+
+def get_label(line: str) -> str:
+    """
+    Returns the label from a line (if it has one, if not returns None)
+
+    >>> get_label('test DC.B $0A')
+    'test'
+
+    >>> get_label(';test DC.B $0A')  # should return nothing
+
+
+    >>> get_label('    BEQ test')  # should return nothing
+
+
+    :param line: The line to get the label from
+    :return: The label in the line
+    """
+
+    if not has_label(line):
+        return None
+
+    stripped = strip_comments(line)
+
+    label = ''
+
+    for c in stripped:
+        if c == ' ':
+            break
+
+        label += c
+
+    return label
+
+
+def strip_label(line: str) -> str:
+    """
+    Strips the label from a line, isolating the rest of the line
+    (side effect: also strips comments)
+
+    >>> strip_label('ORG start')
+    'start'
+
+    >>> strip_label('RTS ;comm')
+    ''
+
+    >>> strip_label(';all commented')
+    ''
+
+    >>> strip_label('MOVE D0, D1 * Moves D0 into D1')
+    'D0, D1 '
+
+    :param line: The line to strip the label from
+    :return: The stripped line
+    """
+
+    stripped = strip_comments(line)
+    if not stripped.strip():  # This line is literally empty after removing comments
+        return ''
+
+    to_return = ''
+    found_space = False
+    found_next = False
+
+    for c in stripped:
+        if c == ' ' and not found_space:
+            found_space = True
+            continue
+
+        if c != ' ' and found_space and not found_next:
+            found_next = True
+
+        if found_next:
+            to_return += c
+
+    return to_return
+
+
+def get_opcode(line: str) -> str:
+    """
+    Gets the opcode out of a line (with or without label)
+
+    >>> get_opcode('start EQU $400')
+    'EQU'
+
+    >>> get_opcode('    MOVE.B D0, D1')
```

Comment:
how are we defining what an indent is? tabs? # of spaces?

Reply:
It can be any number of spaces > 0 (this is how EASy68k handles it, so I'm going off of that)